src/integration_tests/test_with_instance.py

```diff
@@ -0,0 +1,128 @@
+            json.dump(self.state, open(self.statefile, 'w'))
+        if self.cleanup:
+            print('cleaning up')
+            for stackname in self.stacknames:
+                cfn.ensure_destroyed(stackname)
+        print('removing temp dir')
+        # self.rm_temp_dir()
+        # self.assertFalse(os.path.exists(self.temp_dir), "failed to delete path %r in tearDown" % self.temp_dir)
+
+    def test_bootstrap_idempotence(self):
+        "the same stack cannot be created multiple times"
+        bootstrap.create_stack(self.stackname)
+
+    def test_bootstrap_wait_until_in_progress(self):
+        bootstrap._wait_until_in_progress(self.stackname)
+        bootstrap.setup_ec2(self.stackname, self.context)
+
+    def test_bootstrap_update_template_no_updates(self):
+        "a template with no changes can be updated with no problems"
+        bootstrap.update_template(self.stackname, json.load(open(self.cfn_template, 'r')))
+
+    def test_bootstrap_run_script(self):
+        with core.stack_conn(self.stackname, username=BOOTSTRAP_USER):
+            bootstrap.run_script('test.sh')
+
+    def test_core_find_ec2_instances(self):
+        self.assertEqual(len(core.find_ec2_instances(self.stackname)), 1) # 1 node is running
+        self.assertEqual(len(core.find_ec2_instances(self.stackname, state='stopped', allow_empty=True)), 0) # 0 nodes are stopped
+
+    def test_bootstrap_template_info(self):
+        resp = bootstrap.template_info(self.stackname)
+        expecting = {
+            'stack_name': self.stackname,
+            'stack_id': 'arn:aws:cloudformation:us-east-1:512686554592:stack/dummy1--luke-20180420025657-458624/afc10200-4459-11e8-8c52-500c2864aa35',
+            'outputs': {
+                'PublicIP1': '34.229.140.136',
+                'AZ1': 'us-east-1d',
+                'InstanceId1': 'i-027c0ae8fd9054cff',
+                'PrivateIP1': '10.0.2.237'
+            }
+        }
+        self.assertTrue(utils.hasallkeys(resp, expecting.keys()))
+        self.assertTrue(utils.hasallkeys(resp['outputs'], expecting['outputs'].keys()))
+
+    def test_bootstrap_write_environment_info(self):
+        with core.stack_conn(self.stackname, username=BOOTSTRAP_USER):
+            bootstrap.write_environment_info(self.stackname, overwrite=False)
+            bootstrap.write_environment_info(self.stackname, overwrite=True)
+
+    def test_delete_stack_file_while_still_active(self):
+        self.assertEqual(None, bootstrap.delete_stack_file(self.stackname))
+
+    def test_core_describe_stack(self):
+        core.describe_stack(self.stackname)
+
+    def test_core_stack_is(self):
+        self.assertTrue(core.stack_is_active(self.stackname))
+        self.assertFalse(core.stack_is(self.stackname, ['fubar']))
+        self.assertRaises(RuntimeError, core.stack_is, self.stackname, ['fubar'], core.ACTIVE_CFN_STATUS)
+
+    def test_core_active_aws_stacks(self):
+        self.assertTrue(len(core.active_aws_stacks(self.region)) >= 1)
```

Comment:
these assertions are a bit weak, we could check for the presence of the stack in the list instead

Reply:
agreed, I'll do that

Reply:
looking at the output of active_aws_stacks, this test is especially weak and embarrassing. for some reason I must have assumed the test stack was the only one being brought up